### PR TITLE
Osio reuse

### PR DESCRIPTION
**Pull request by edewit**
In order to reuse the Identity in OSIO we need to move this logic to the services

### CODE REVIEW

File: services/keycloak-service-api/src/main/java/io/fabric8/launcher/service/keycloak/api/KeycloakService.java [hunk 1]
```diff
@@ -14,10 +14,11 @@
     /**
      * Returns the Openshift v3 {@link Identity} used for authenticating with the Openshift console
      *
-     * @param token the keycloak access token
+     * @param authorization authorization header
+     * @param cluster the cluster to use to get the token for
      * @return the openshift v3 token assigned to the given keycloak access token
      */
-    Identity getOpenShiftIdentity(String token);
+    Identity getOpenShiftIdentity(String authorization, String cluster);
```
Comment (gastaldi) [inline services/keycloak-service-api/src/main/java/io/fabric8/launcher/service/keycloak/api/KeycloakService.java]: The whole `Authorization` header including the `Bearer: ` prefix?

File: web/src/main/java/io/fabric8/launcher/web/api/OpenShiftResource.java [hunk 2]
```diff
@@ -61,21 +61,23 @@
...
         Set<OpenShiftCluster> clusters = clusterRegistry.getClusters();
-        if (request.getParameterMap().containsKey("all") || useDefaultIdentities()) {
+        if (request.getParameterMap().containsKey("all") || IdentityFactory.useDefaultIdentities()) {
             // Return all clusters
             clusters
                     .stream()
                     .map(OpenShiftCluster::getId)
                     .forEach(arrayBuilder::add);
         } else {
-            final KeycloakService keycloakService = this.keycloakServiceInstance.get();
```
Comment (gastaldi) [inline web/src/main/java/io/fabric8/launcher/web/api/OpenShiftResource.java]: Why was this line removed?

Comment (edewit) [reply web/src/main/java/io/fabric8/launcher/web/api/OpenShiftResource.java]: cause `keycloakService` is a field now

File: web/src/main/java/io/fabric8/launcher/web/api/OpenShiftResource.java [hunk 3]
```diff
@@ -61,21 +61,23 @@
     @Inject
     private OpenShiftClusterRegistry clusterRegistry;
 
+    @Inject
+    private KeycloakService keycloakService;
```
[the remark this answers is not in the document]
Comment (edewit) [reply web/src/main/java/io/fabric8/launcher/web/api/OpenShiftResource.java]: I changed it so you can, the check to see if the keycloakUrl is set is now moved from the constructor to the method that uses the keycloak url in order to use the keycloakService to get the identity in case there is no keycloak

**Comment by gastaldi** [comment]
Introduced the `Identities` service in `core` to handle identities easily.  Closing PR now
